File: assignments/a1/q7/Python/Balanced.py
#Balanced:top#
#Balanced:import#

# <balanced> ::= <bal_paren> AT
class Balanced(_Start): #Balanced:class#

    className = "Balanced"
    ruleString = "<balanced> ::= <bal_paren> AT"
    bal_paren = None

    def __init__(bal_paren):
        #Balanced:init#
        self.bal_paren = bal_paren

#Balanced#

File: assignments/a1/q7/Java/Balanced.java
//Balanced:top//
//Balanced:import//
import java.util.*;

// <balanced> ::= <bal_paren> AT
public class Balanced extends _Start /*Balanced:class*/ {

    public static final String $className = "Balanced";
    public static final String $ruleString =
        "<balanced> ::= <bal_paren> AT";

    public Bal_paren bal_paren;

    public Balanced(Bal_paren bal_paren) {
//Balanced:init//
        this.bal_paren = bal_paren;
    }

    public static Balanced parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<balanced>", scn$.lno);
        Bal_paren bal_paren = Bal_paren.parse(scn$, trace$);
        scn$.match(Token.Match.AT, trace$);
        return new Balanced(bal_paren);
    }

//Balanced//
}
